tests/sp2n/Test_project_on_Sp.cc

```diff
@@ -55,35 +60,119 @@ int main (int argc, char **argv)
     
     std::cout << GridLogMessage << "Unitary matrix deformed " << std::endl;
     std::cout << GridLogMessage << "now U adjU - 1 = " << norm2(aux) << std::endl;
+    
     std::cout <<GridLogMessage << std::endl;
-
+    std::cout << GridLogMessage << "#   #   #   #" << std::endl;
+    std::cout << GridLogMessage << "Group" << std::endl;
+    std::cout << GridLogMessage << "#   #   #   #" << std::endl;
     std::cout <<GridLogMessage << std::endl;
-    std::cout << GridLogMessage << "Projecting on Sp2n " << std::endl;
-
+    //  Testing ProjectOnSpGroup
+    std::cout << GridLogMessage << "Testing ProjectOnSpGroup" << std::endl;
+    std::cout << GridLogMessage << "Apply ProjectOnSpGroup to deformed matrix" << std::endl;
     U = ProjectOnSpGroup(U);
-    //U = ProjectOnGroup(U);
+
     aux = U*adj(U) - identity;
-    std::cout <<GridLogMessage << std::endl;
-    std::cout <<GridLogMessage << std::endl;
-    std::cout << GridLogMessage << "Unitary check after Sp(2n) projection " << std::endl;
+    std::cout << GridLogMessage << "Unitary check after ProjectOnSpGroup " << std::endl;
     std::cout << GridLogMessage << "U adjU - 1 = " << norm2(aux) << std::endl;
     assert( norm2(aux) < 1e-8);
-    std::cout <<GridLogMessage << std::endl;
-    std::cout <<GridLogMessage << std::endl;
-    
-    // checks on determinant
-    std::cout << GridLogMessage << "Det after Projection on Sp2n = " << norm2( Determinant(U) ) / vol << std::endl;
-    std::cout <<GridLogMessage << std::endl;
-    std::cout <<GridLogMessage << std::endl;
     
     // actual sp2n check
-    std::cout << GridLogMessage << "Checking invariance after projection "<< std::endl;
-    Sp<Nc>::OmegaInvariance(U);
+    std::cout << GridLogMessage << "Checking Omega invariance after ProjectOnSpGroup" << std::endl;
+    Sp<Nc>::OmegaInvariance(U);     // no assertion here, but the next check will kill us if we are not simplectic
```

Comment:
Thanks for the comment! Even better: Make `OmegaInvariance` actually return something (e.g. `sdiff`) and actually do an assertion.